src/main/java/persistence/sql/dml/InsertQueryBuilder.java

```diff
@@ -1,53 +1,72 @@
+import java.lang.reflect.InvocationTargetException;
 import java.util.Arrays;
 import java.util.List;
 
 import static persistence.sql.common.SqlConstant.CLOSING_PARENTHESIS;
 import static persistence.sql.common.SqlConstant.COMMA;
 
-public class InsertQueryBuilder<T> {
+public class InsertQueryBuilder {
     public static final String INSERT_QUERY_START = "INSERT INTO %s (";
     public static final String VALUES = " VALUES (";
-    private final Table table;
+    private final TableClause tableClause;
 
     public InsertQueryBuilder(Class<?> entity) {
         if (!entity.isAnnotationPresent(Entity.class)) {
             throw new InvalidEntityException();
         }
-        this.table = new Table(entity);
+        this.tableClause = new TableClause(entity);
     }
 
-    public String getInsertQuery(T target) {
+    public String getInsertQuery(Object instance) {
 
-        List<Field> fields = Arrays.stream(target.getClass().getDeclaredFields()).toList();
+        List<Field> fields = Arrays.stream(instance.getClass().getDeclaredFields()).toList();
 
-        return String.format(INSERT_QUERY_START, table.name()) +
-                String.join(COMMA, table.columnNames()) +
+        return String.format(INSERT_QUERY_START, tableClause.name()) +
+                String.join(COMMA, tableClause.columnNames()) +
                 CLOSING_PARENTHESIS +
                 VALUES +
-                String.join(COMMA, new Values(fields, target).getQueries()) +
+                String.join(COMMA, new ValueClauses(fields, instance).getQueries()) +
                 CLOSING_PARENTHESIS;
     }
 
-    public String getInsertQuery(T target, Columns colums, Values values) {
+    public String getInsertQuery(List<String> columnNames, List<Object> columValues) {
 
-        List<Field> fields = Arrays.stream(target.getClass().getDeclaredFields()).toList();
+        if (columnNames.size() != columValues.size()) {
+            throw new InvalidValueClausesException();
+        }
 
-        return String.format(INSERT_QUERY_START, table.name()) +
-                String.join(COMMA, colums.getQueries()) +
-                CLOSING_PARENTHESIS +
-                VALUES +
-                String.join(COMMA, values.getQueries()) +
-                CLOSING_PARENTHESIS;
+        Object instance;
+        try {
+            instance = initInstance(columnNames, columValues);
+        } catch (NoSuchFieldException | NoSuchMethodException | IllegalAccessException | InvocationTargetException |
+                 InstantiationException e) {
+            throw new IllegalArgumentException("잚못된 요청입니다.");
+
+        }
+        return this.getInsertQuery(instance);
+    }
+
+    private Object initInstance(List<String> columnNames, List<Object> columValues)
+            throws InstantiationException, IllegalAccessException, InvocationTargetException, NoSuchMethodException,
+            NoSuchFieldException {
+        Object instance;
+        instance = tableClause.newInstance();
```

Comment:
```suggestion
        Object instance;
        instance = tableClause.newInstance();
```
별다른 로직이 없는데 바로 초기화를 하지않으면 코드를 보는 동료 개발자가 실수를 할수 있는 길을 열어주거나 머릿속을 복잡하게 만들지 않을까요??

Reply:
저는 변수 초기화를 말씀드린 부분입니다. 😄 
`Object instance = tableClause.newInstance();`